02-oop-inheritance/functions.js

```diff
@@ -0,0 +1,108 @@
+var Movie = function (title, duration, director) {
+	
+		this.title = title;
+		this.duration = duration;
+		this.director = director;
+		
+
+
+};
+
+Movie.prototype = {
+	
+
+
+	play:function(){ MovieObserver.playing(this.title); },
+	stop:function(){ MovieObserver.stopping(this.title); },
+	set:function(attr, value){ this[attr] = value; },
+	get:function(attr){
+		console.log(this[attr]); }
+};
+
+// Observer
+
+var MovieObserver = (function () {
+	var playing = function (name) {
+		console.log("Playing " + name + "...");
+	},
+	stopping = function (name) {
+		console.log("Stopping " + name + "...");
+	};
+
+	return {playing:playing, stopping:stopping};
+
+})();
+
+
+
+// New object + extends
+
+var DownloadableMovie = function (title, duration, director, link){
+	Movie.call(this,title, duration, director);
+	this.link = link;
+
+};
+
+DownloadableMovie.prototype = Object.create(Movie.prototype);
```

Comment:
Good, but careful, Object.create is an ES5 method, this won't work in IE8-. You might need a polyfill in a real project.